image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
 
-Of course any downstream consumer who chooses to use "foo:latest" will be taking on the risk of any incompatible changes introduced.
+In addition, use tags in the `FROM` instruction (for example,  `rhel:rhel7`) to make it clear to users exactly which version of an image your image is based on. Using a tag other than `latest` ensures your image is not subjected to breaking changes that might go into the `latest` version of an upstream image.
 
+*Tagging Images*
 
-=== Do not start multiple processes
-It is not a good idea to start multiple services (like database, sshd, …​) inside one container. Containers are lighweight and can be easily linked together for multiprocess orchestration.  OpenShift allows you to easily colocate and comanage related images by grouping them into a single pod.  This will ensure they share a network namespace and storage for communication.  It also makes updates less disruptive as each image can be updated less frequently and independently.  Having a single process also makes signal handling flows clearer as you do not need to manage routing signals to spawned processes.
+When tagging your own images, strive to maintain backwards compatibility within a tag. For example, if you provide an image named [sysitem]#foo# and it currently includes version 1.0, you might provide a tag of _foo:v1_. When you update the image, as long as it continues to be compatible with the original image, you can continue to tag the new image _foo:v1_, and downstream consumers of this tag will be able to get updates without being broken.
 
+If you later release an incompatible update, then you should switch to a new tag, for example _foo:v2_. This allows downstream consumers to move up to the new version at will, but not be inadvertently broken by the new incompatible image. Of course, any downstream consumer who chooses to use _foo:latest_ will be taking on the risk of any incompatible changes introduced.
 
-=== Use exec in wrapper scripts
-See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `exec` in Wrapper Scripts_
+*Avoiding Multiple Processes*
 
-Also be aware that your process runs as PID 1 when running in a Docker container.  This means that if your main process terminates, the entire container will be stopped, killing any child processes you may have launched from your PID 1 process.  Additional implications to be aware of are documented http://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/[here].  For a deep dive on PID 1 and init systems, you can read https://felipec.wordpress.com/2013/11/04/init/[this] as well.
+Do not start multiple services, such a database and [sysitem]#sshd#, inside one container. Containers are lightweight and can be easily linked together for multiprocess orchestration. OpenShift allows you to easily collocate and comanage related images by grouping them into a single pod.
 
```

Comment:
[Typo]
s/such a/such as

[Suggestion]
We recommend that you avoid running multiple services, such as a database and [sysitem]#sshd#, inside one container. This is not necessary because containers are lightweight and can be easily linked together for orchestrating multiple processes.